operator/apis/resources/v2/user_types.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v2
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/gravitational/teleport/api/types"
+)
+
+const DescriptionKey = "description"
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// UserSpec defines the desired state of User
+type UserSpec types.UserSpecV2
+
+// UserStatus defines the observed state of User
+type UserStatus struct {
+	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// User is the Schema for the users API
+type User struct {
```

Comment:
Yeah, I'm not sure it would work out without some core changes

There's also another thing: we are already `Check` and `SetDefaults` on the Teleport API
Having two places calling the same code in the same request doesn't seem the right thing to do given those applications are meant to be side by side (in a sidecar architecture)

Reply:
Yeah, that's fair. I was just wondering about feedback to the user. If they are mainly configuring it via k8s custom resources, it might be nice if the defaults were reflected back to them after submitting the manifest. Let's revisit this at a later date though, I think it's fine as it is.